src/mactelnetd.c

```diff
@@ -634,12 +646,44 @@ static void handle_data_packet(struct mt_connection *curconn, struct mt_mactelne
 				memset(curconn->trypassword, 0, sizeof(curconn->trypassword));
 			}
 			init_packet(&pdata, MT_PTYPE_DATA, pkthdr->dstaddr, pkthdr->srcaddr, pkthdr->seskey, curconn->outcounter);
-			plen = add_control_packet(&pdata, MT_CPTYPE_PASSSALT, (curconn->pass_salt), 16);
+			plen = add_control_packet(&pdata, MT_CPTYPE_PASSSALT, (curconn->pass_salt), use_md5 ? 16 : 49);
```

Comment:
This is guarded by `if (cpkt.cptype == MT_CPTYPE_BEGINAUTH && use_md5 == 1) {` at L638 (the ` && use_md5 == 1` was added with this patch), and thus it seems to me that `use_md5 ? 16 : 49` (which was also added in this patch) always evaluates to 16 here. Thus I wonder what is the goal here? Probably either the `if` or this line needs to change.

Reply:
Yes, this should probably be 16, as use_md5 is handeled in the `else if` below.